File: pkg/models/auth_response.go
package models

// AuthResponse returns a view-object of user authorization
type AuthResponse struct {
	Token               string         `json:"token"`
	Need2FA             bool           `json:"need2FA"`
	User                *UserBasicInfo `json:"user"`
	NotificationContent string         `json:"notificationContent,omitempty"`
}

// RegisterResponse returns a view-object of user register response
type RegisterResponse struct {
	AuthResponse
	NeedVerifyEmail       bool `json:"needVerifyEmail"`
	PresetCategoriesSaved bool `json:"presetCategoriesSaved"`
}

File: src/models/auth_response.ts
import type { ApplicationCloudSetting } from '@/core/setting.ts';

import type { UserBasicInfo } from './user.ts';

export interface AuthResponse {
    readonly token: string;
    readonly need2FA: boolean;
    readonly user?: UserBasicInfo;
    readonly applicationCloudSettings?: ApplicationCloudSetting[];
    readonly notificationContent?: string;
}

export interface RegisterResponse extends AuthResponse {
    readonly needVerifyEmail: boolean;
    readonly presetCategoriesSaved: boolean;
}
